File: Introduccion a Poo/Ejercicio 8/Java/Banco.java
import java.util.ArrayList;
public class Banco {
    private ArrayList<Usuario> usuarios = new ArrayList<>();
    public void agregarUsuario(Usuario u) {
        if (usuarios.size() < 10) {
            usuarios.add(u);
        }
    }
    public void depositar(String nombre, double cantidad) {
        for (Usuario u : usuarios) {
            if (u.getNombre().equals(nombre)) {
                u.depositar(cantidad);
            }
        }
    }
    public void retirar(String nombre, double cantidad) {
        for (Usuario u : usuarios) {
            if (u.getNombre().equals(nombre)) {
                u.retirar(cantidad);
            }
        }
    }
    public void darBonos() {
        for (Usuario u : usuarios) {
            u.darBono();
        }
    }
    public void totalBanco() {
        double total = 0;
        for (Usuario u : usuarios) {
            total += u.getSaldo();
        }
        System.out.println("Total del banco: " + total);
    }
}
public class Usuario {
    private String nombre;
    private double saldo;
    public Usuario(String nombre, double saldo) {
        this.nombre = nombre;
        this.saldo = saldo;
    }
    public String getNombre() {
        return nombre;
    }
    public double getSaldo() {
        return saldo;
    }
    public void depositar(double cantidad) {
        saldo += cantidad;
    }
    public void retirar(double cantidad) {
        saldo -= cantidad;
    }
    public void darBono() {
        saldo *= 1.02;
    }
}

File: Introduccion a Poo/Ejercicio 8/Python/Banco.py
class Usuario:
    def __init__(self, nombre, s):
        self.nombre = nombre
        self.s = s
class Banco:
    def __init__(self):
        self.usuarios = []
    def agregar(self, u):
        if len(self.usuarios) < 10:
            self.usuarios.append(u)
    def retirar(self, nombre, cantidad):
        for u in self.usuarios:
            if u.nombre == nombre:
                u.s = u.s - cantidad
    def depositar(self, nombre, cantidad):
        for u in self.usuarios:
            if u.nombre == nombre:
                u.s = u.s + cantidad
    def bono(self):
        for u in self.usuarios:
            u.s = u.s * 1.02
    def total(self):
        t = 0
        for u in self.usuarios:
            t = t + u.s
        print("Dinero total en el banco:", t)
b = Banco()
u1 = Usuario("Ana", 100)
u2 = Usuario("Luis", 200)
b.agregar(u1)
b.agregar(u2)
b.depositar("Ana", 50)
b.retirar("Luis", 30)
b.bono()
b.total()